Add __DIR__ to fix not launched as bin/imbolauncher

matslindh opened the pull request:
If imbolauncher is invoked outside of its own homedir (for example through an absolute path or inside the `bin` directory itself), the existing require will fail - as it uses a relative import from the CWD.

matslindh committed:
Add __DIR__ to fix not launched as bin/imbolauncher

christeredvartsen reviewed (changes requested):
When installing `imbo/imbolauncher` using composer, as in having `imbo/imbolauncher` as a requirement, composer will create a symlink to `../imbo/imbolauncher/bin/imbolauncher` in the `vendor/bin` directory.

The current path will work both on a local installation, and when installed in the above fashion.

By adding the `__DIR__ . '/../` part only the local installation will work as installed packages don't get their own vendor directories.

matslindh committed:
Search for the first vendor/ dir for dependencies

matslindh commented:
I've updated the PR by searching for a vendor dir, as this will work regardless of the CWD when the command is invoked, as long as the dependencies has been installed (either after a git clone, after a require: in a composer.json or part of a create-project composer command). I still want to test this fix on Windows before making it final.

matslindh commented:
Works properly on windows as well.

matslindh committed:
Add fix for other os' and error message

By testing for the previous entry instead of hardcoding what we're looking
for, we can avoid any non-terminating loops in a general way.

This also adds an error message about running composer install from the
imbolauncher directory if the dependencies are missing.

matslindh committed:
Add "--no-dev" to example composer cmd

matslindh commented:
(at least now it does)
